src/storage/txn/scheduler.rs

```diff
@@ -687,15 +693,14 @@ impl<E: Engine, L: LockManager> TxnScheduler<E, L> {
             .collect();
 
         let cmd = commands::AcquirePessimisticLockResumed::from_lock_wait_entries(awakened_entries);
+        let tag = cmd.cmd.tag();
         let cid = specified_cid.unwrap_or_else(|| self.inner.gen_id());
-        let task = Task::new(cid, cmd.into());
-
-        // TODO: Make flow control take effect on this thing.
-        self.schedule_command(
-            task,
-            SchedulerTaskCallback::LockKeyCallbacks(key_callbacks),
-            prepared_latches,
-        );
+        let callback = SchedulerTaskCallback::LockKeyCallbacks(key_callbacks);
+        if let Ok(task) = Task::allocate(cid, cmd.into(), self.inner.memory_quota.clone()) {
+            self.schedule_command(task, callback, prepared_latches);
+        } else {
+            Self::fail_with_busy(tag, callback);
```

Comment:
I'm still trying to remember the details here though... it looks not sufficient to do this for the failure case. At lease the `prepared_latches` needs to be released.

Reply:
If it's called from `wake_up_legacy_pessimistic_locks` and the resumed pessimistic lock command failed to be scheduled, it seems we need some extra work to guarantee that later waiting entries, if any, should be able to be waken up later when it can be allowed by the flow contro. Anyway it's not as easy as simply cancel it when flow control happens.

Reply:
I see, perhaps we could do this related refactor in later PRs and force to schedule command here.

Is there similar risks to cancel the re-scheduled command in `on_read_finish` and `on_write_finish`?